addons/l10n_nl/models/account_journal.py

```diff
@@ -6,6 +6,17 @@
 class AccountJournal(models.Model):
     _inherit = 'account.journal'
 
+    @api.model
+    def _fill_missing_values(self, vals):
+        super()._fill_missing_values(vals)
+
+        if vals.get('type') != 'purchase':
+            return
+
+        company = self.env['res.company'].browse(vals['company_id']) if vals.get('company_id') else self.env.company
+        if company.country_id.code == "NL":
+            vals['type_control_ids'] = [self.env.ref('account.data_account_type_direct_costs').id]
```

Comment:
If there are already values for this key, you should add them to the id you are adding

Reply:
Done